src/NuGet.Core/NuGet.PackageManagement/NuGetProjectAction.cs

```diff
@@ -52,7 +55,14 @@ public class NuGetProjectAction
         /// </summary>
         public NuGetProject Project { get; private set; }
 
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads.
```

Comment:
This should go in the GlobalSuppressions file with an entry for this particular class.

Reply:
I was the one who suggested to @martinrrm to put it in the file, not in the GlobalSupressions.cs.

The reason is when I'm making changes to a file for some other bug fix or feature, if I see there are supressions in the code, then while I'm making my bugfix/feature, I can fix the code analysis warning at the same time.  When the supressions are in the GlobalSupressions.cs file, I don't even know that there's an opportunity to improve the code quality.

I'm in team anti-GlobalSupressions.cs.

Reply:
@zivkan I can agree with that! It's harder to track down the suppressions if they're in a separate file and as a result, they tend to live longer as suppressions rather than be fixed.

Reply:
There are definitely different opinions on the matter something we discussed when @kartheekp-ms originally added them. 

If you're using VisualStudio, GlobalSupressions show up as references.